doc/design/upgrade.md

```diff
@@ -125,15 +135,114 @@ Here is the upgrade workflow
 - Cleanup env after upgrade (remove upgrade snapshot, remove uploaded iso files if upgrade failed)
 
 This ensures that a failure to upload ISOs for whatever reason is detected before we take down the existing version. It limits the failure modes after shutting down the appliance to:
+
 1. failed to update appliance configuration
 2. failed to power on the appliance (e.g. system resource constraints)
 3. failed to boot appliance
 4. failed to acquire network addresses (this has been seen in real world examples - we may wish to attempt to preserve/reuse the IPs the appliance had prior to update, which should still be present in the extraconfig)
 5. failed to rebuild system state from vsphere infrastructure
 
+## Design - Phase2
+
+Although we did lots of code refactor for VCH appliance configuration, it's still unavoidable to continue changing that structure. To make sure we don't break  upgrade after GA, we need a solution for the data migration anyway.
+
+- Appliance guestinfo, which is used to persist VCH appliance configuration
+- Container guestinfo, which is used to persist container configuration
+- KeyValue Store, which is used to persist image metadata and network portlayer data
+- Appliance log files
+- vSphere object management logic change
+
+### Guestinfo Migration
+
+- One VirtualContainerHost structure per one version, this will be used by extraconfig package to read back old vch's configuration
```

Comment:
This is a good intention towards to leverage portlayer for VCH appliance provisioning. Let me do it now. It's good to have some roughly idea before go ahead with the upgrade approach.